File: wheystore/wheystore.Web/Modules/SanPhamDB/KhachHang/KhachHangColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.KhachHang")]
[BasedOnRow(typeof(KhachHangRow), CheckNames = true)]
public class KhachHangColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public int UserId { get; set; }
    [EditLink]
    public string HoTen { get; set; }
    public string DienThoai { get; set; }
    public string Email { get; set; }
    public string DiaChi { get; set; }

    [LookupEditor(typeof(TinhThanhRow)), QuickFilter, DisplayName("Tỉnh Thành")]
    public int? TinhThanhId { get; set; }

    

    [DisplayName("Tỉnh Thành")]
    public string TinhThanhTen { get; set; }

    [DisplayName("Quận/Huyện")]
    public string QuanHuyenTen { get; set; }

    [DisplayName("Xã/Phường")]
    public string XaPhuongTen { get; set; }

    public string MaBuuChinh { get; set; }
    public DateTime NgaySinh { get; set; }


    [DisplayName("Giới Tính")]
    public short? GioiTinh { get; set; }

    
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/KhachHangColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { KhachHangRow } from "./KhachHangRow";

export interface KhachHangColumns {
    Id: Column<KhachHangRow>;
    UserId: Column<KhachHangRow>;
    HoTen: Column<KhachHangRow>;
    DienThoai: Column<KhachHangRow>;
    Email: Column<KhachHangRow>;
    DiaChi: Column<KhachHangRow>;
    TinhThanhId: Column<KhachHangRow>;
    QuanHuyenId: Column<KhachHangRow>;
    XaPhuongId: Column<KhachHangRow>;
    TinhThanhTen: Column<KhachHangRow>;
    QuanHuyenTen: Column<KhachHangRow>;
    XaPhuongTen: Column<KhachHangRow>;
    MaBuuChinh: Column<KhachHangRow>;
    NgaySinh: Column<KhachHangRow>;
    GioiTinh: Column<KhachHangRow>;
}

export class KhachHangColumns extends ColumnsBase<KhachHangRow> {
    static readonly columnsKey = 'SanPhamDB.KhachHang';
    static readonly Fields = fieldsProxy<KhachHangColumns>();
}